File: evaluations/powersimulations/verify_install.jl
"""
Verify PowerSimulations.jl installation by loading IEEE 39-bus case
and running DC power flow via PowerFlows.jl.
"""

using PowerSystems
using PowerFlows
using PowerNetworkMatrices

data_dir = joinpath(@__DIR__, "..", "..", "data", "networks")
case_file = joinpath(data_dir, "case39.m")

# Load system from MATPOWER file
sys = System(case_file)
println("System: $(get_name(sys))")
println("Buses: $(length(collect(get_components(ACBus, sys))))")

# Run DC power flow
result = solve_powerflow(DCPowerFlow(), sys)
println("DC power flow completed: $(result)")

exit(0)

File: evaluations/pypsa/verify_install.py
"""Verify PyPSA installation by running DC power flow on IEEE 39-bus case."""

from __future__ import annotations

import sys
from pathlib import Path

import pypsa
from matpowercaseframes import CaseFrames

DATA_DIR = Path(__file__).resolve().parent.parent.parent / "data" / "networks"

# Load case39.m via matpowercaseframes → pypower ppc dict
cf = CaseFrames(str(DATA_DIR / "case39.m"))
ppc = {
    "version": "2",
    "baseMVA": cf.baseMVA,
    "bus": cf.bus.values,
    "gen": cf.gen.values,
    "branch": cf.branch.values,
}

# Import into PyPSA and run DC power flow
net = pypsa.Network()
net.import_from_pypower_ppc(ppc)
net.lpf()

print(f"PyPSA version: {pypsa.__version__}")
print(f"Buses: {len(net.buses)}")
print(f"Lines: {len(net.lines)}")
print("DC power flow completed successfully")
sys.exit(0)
